siliconcompiler/design.py

```diff
@@ -0,0 +1,535 @@
+            # TODO: add source info for comments to flist.
+            with open(filename, "w") as f:
+                for i in fileset:
+                    if not isinstance(i, str):
+                        raise ValueError("fileset key must be a string")
+                    for j in ['idir', 'define', 'file']:
+                        if j == 'idir':
+                            vals = self.get('fileset', i, 'idir')
+                            cmd = "+incdir+"
+                        elif j == 'define':
+                            vals = self.get('fileset', i, 'define')
+                            cmd = "+define+"
+                        else:
+                            vals = self.get('fileset', i, 'file', 'verilog')
+                            cmd = ""
+                        if vals:
+                            for item in vals:
+                                f.write(f"{cmd}{item}\n")
+        else:
+            raise ValueError(f"{fileformat} is not supported")
+
+    ################################################
+    def read_fileset(self, fileset: str, filename: str, fileformat=None) -> None:
+        """Imports filesets from a standard formatted text file.
+
+        Currently supports Verilog `flist` format only.
+        Intended to support other formats in the future.
+
+        Args:
+            fileset (str or list[str]): Fileset(s) to import.
+            filename (str or Path): Output file name.
+            fileformat (str, optional): Export format.
+                Inferred from file extension if not given.
+
+        """
+
+        if filename is None:
+            raise ValueError("read_fileset() filename cannot be None")
+
+        if not fileformat:
+            formats = {}
+            formats['f'] = 'flist'
+            fileformat = formats[Path(filename).suffix.strip('.')]
+
+        if fileformat == "flist":
+            raise RuntimeError("read_fileset is not implemented yet")
+        else:
+            raise ValueError(f"{fileformat} is not supported")
+
+    ################################################
+    # Helper Functions
+    ################################################
+    def _set(self, fileset, option, value, typelist=None):
+        '''Sets a parameter value in schema.
+        '''
+
+        # check for a legal fileset
+        if not isinstance(fileset, str):
+            raise ValueError("fileset key must be a string")
+
+        # Check for legal types
+        legalval = False
+        for item in typelist:
+            if isinstance(value, item) and not isinstance(value, tuple):
+                legalval = True
+        if not legalval:
+            raise ValueError("value type must be str or List")
+
+        # None is illegal for all setters
+        if value is None:
+            raise ValueError(f"None is an illegal {option} value")
+
+        self.set('fileset', fileset, option, value)
+        return self.get('fileset', fileset, option)
```

Comment:
any reason to not just return the result of self.set() directly?

Reply:
I forgot that you return the value, will do